api/modules/files/model.js

```diff
@@ -1,28 +1,31 @@
-var BaseModel = require('../../classes/base_model');
+"use strict";
 
-var Projects = require('../projects/model');
-var Users = require('../users/model');
+const BaseModel = require(`../../classes/base_model`);
 
-var instanceProps = {
-  tableName: 'files',
-  project: function() {
-    return this.belongsTo(Projects);
+const ProjectsModel = require(`../projects/model`);
+
+const instanceProps = {
+  tableName: `files`,
+  project() {
+    return this.belongsTo(ProjectsModel);
   },
-  user: function() {
-    return this.belongsTo(Users).through(Projects);
+  user() {
+    // We require in the function as opposed to adding a top-level require
+    // to resolve the circular dependencies between models
+    return this.belongsTo(require(`../users/model`)).through(ProjectsModel);
```

Comment:
yes...it's a very annoying bug! Requiring this at runtime fixes a circular dependency so that the exports happens first before the require.